ndindex/tests/helpers.py

```diff
@@ -24,12 +24,14 @@
 def prod(seq):
     return reduce(mul, seq, 1)
 
+positive_ints = integers(1, 10)
 nonnegative_ints = integers(0, 10)
 negative_ints = integers(-10, -1)
 ints = lambda: one_of(negative_ints, nonnegative_ints)
+ints_nonzero = lambda: one_of(negative_ints, positive_ints)
 
 def slices(start=one_of(none(), ints()), stop=one_of(none(), ints()),
-           step=one_of(none(), ints())):
+           step=one_of(none(), ints_nonzero())):
```

Comment:
This includes zero step because it is used to test the slice constructor. Although it would probably be better to use this for the tests that don't actually test that (see the discussion at https://github.com/Quansight/ndindex/issues/46). It may be better to work on this in a separate PR, though.

Reply:
I added this since I kept getting `ValueErrors` due to the related check in the `Slice` constructor.

So what I don't get is how this wasn't causing errors previously for tests using the `@given(ndindices)` strategy. Wouldn't that just mean that (due to one of the many weird quirks of hypothesis) `@given(ndindices)` just happened to never attempt to generate `ndindex(slice(x, y, 0))`?

Reply:
ndindices uses `filter(_doesnt_raise)` which make it only generate valid indices. Actually I suspect this means several tests